src/main/java/edu/ie3/datamodel/io/source/PowerValueSource.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * © 2025. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.io.source;
+
+import edu.ie3.datamodel.models.profile.LoadProfile;
+import edu.ie3.datamodel.models.profile.PowerProfile;
+import edu.ie3.datamodel.models.value.PValue;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import java.util.function.Supplier;
+import javax.measure.quantity.Energy;
+import javax.measure.quantity.Power;
+import tech.units.indriya.ComparableQuantity;
+
+/** Interface defining base functionality for power value sources. */
+public sealed interface PowerValueSource<
+        P extends PowerProfile, ID extends PowerValueSource.InputData>
+    permits PowerValueSource.MarkovBased, PowerValueSource.TimeSeriesBased {
+
+  /** Returns the profile of this source. */
+  P getProfile();
+
+  /**
+   * Method to get the next power value based on the provided input data.
+   *
+   * @param data input data that is used to calculate the next power value.
+   * @return an option for the power value.
+   */
+  Optional<PValue> getValue(ID data);
```

Comment:
I think this method now becomes obsolete with the addition of `getValueSupplier`. We can just use a supplier all the time, even if it always returns the same value.